services/checks/checks.go

```diff
@@ -274,16 +301,23 @@ func (s *Service) executeChecks(ctx context.Context) {
 
 // executeMySQLChecks runs MySQL checks for available MySQL services.
 func (s *Service) executeMySQLChecks(ctx context.Context) ([]string, error) {
-	targets, err := s.findTargets(models.MySQLServiceType)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to find proper agents and services")
-	}
-
 	checks := s.getMySQLChecks()
 
 	var res []string
-	for _, target := range targets {
-		for _, c := range checks {
+	for _, c := range checks {
+		minAgentVersion, err := getMinAgentVersionForCheckType(c.Type)
+		if err != nil {
+			s.l.Errorf("failed to get min agents version for type %s, reason: %s", c.Type, err)
+			continue
+		}
+
+		targets, err := s.findTargets(models.MySQLServiceType, minAgentVersion)
```

Comment:
looks like we can just pass c.Type into findTargets and call getMinAgentVersionForCheckType there to not copy-paste it everywhere.

Reply:
Technically yes, but I think it will be monster-complex function. Now it knows only about services\agents, if we move `getMinAgentVersionForCheckType` then it also should know about checks. For me better to have this logic separately 🧐

Reply:
@BupycHuk ^ what do you think?

Reply:
I'm ok with that, but we duplicated this code for each service type and we won't use them independently.

Reply:
I think the best solution could be to store checks with additional data:
```
type checkInfo struct {
  check.Check
  minAgentVersion *version.Parsed
}
…
mySQLChecks      []checkInfo
```

Then findTargets could use that info.

But let's do that during our next refactoring. I think there are other bigger pieces that could be rearranged.